### DISCUSSION TITLE
Generate performance: takes 8 minutes, ~12GB of RAM

### DISCUSSION THREAD

**Discussion by geertjanvdk** [99designs/gqlgen#2792] [Q&A]
I am facing an issue with a medium schema size (about 1200 line; no comments/help; fields on each line), and it takes around 8 minutes to generate. My MacBook Air M2 has 16GB, and I need to shutdown lots to get gqlgen to save a few minutes (would be around 6 minutes).

Monitoring the process it is taking about 12GB. I have 23 direct dependencies.

4 months ago, it was 4 minutes. It's not like the schema gets that bigger all of sudden, but any change adds quite a bit of time..
Any great tools or examples on how to debug this? Any buttons I can turn to tell it to not look too far for things?
The excesive use of RAM is.. quite a thing, no?

(private project; cannot share, but bits I could if needed)

---

Adding a few timers, I get weird things like:

```
bindField           333ns: Mutation.remove******
bindField           334ns: Mutation.save******
bindField           333ns: Mutation.remove******
bindField    2.148426833s: Mutation.createContact
bindField    2.571112333s: Mutation.updateContact
bindField     1.50583325s: Mutation.deleteContact
bindField    2.334608292s: Mutation.update******
bindField        31.625µs: Mutation.delete******
bindField         21.25µs: Mutation.create******
bindField        31.666µs: Mutation.update******
bindField         1.166µs: Mutation.delete******
bindField    3.792987375s: Mutation.create******
bindField    2.711437583s: Mutation.update******
```

I cannot explain these 2/3 seconds for some or most resolvers. Also, adding those up would not get 8 minutes. Digging more..

---

Similar as bug #918

**Comment by UnAfraid** [comment]
I just tested on Macbook Pro M2 with 32 gb of ram on large production project with about 4400 lines of schema.
```bash
bash-3.2$ go version
go version go1.21.0 darwin/arm64
bash-3.2$
bash-3.2$ time go run github.com/99designs/gqlgen --config gqlgen.yml generate
time go run github.com/99designs/gqlgen --config gqlgen.yml generate  17.74s user 5.16s system 265% cpu 8.635 total
```

generate took ~18s
Memory usage shown by activity monitor was about 31 mb

**Reply by geertjanvdk** [reply]
Yeah, it is something on my side.. but what? I am going to try to scale down the project so that there are no more dependencies.. This will cost me much but hey!

I hope(d) that somebody had a similar situation..

Really what can do 12GB :(

**Reply by UnAfraid** [reply]
You could check how much free memory you have, check in activity monitor what's taking your memory or keeping your CPU busy.
Worst case you can probably use CI like GitHub Actions to generate the schema for you for free.

**Reply by geertjanvdk** [reply]
Working around the problem is easy.
I have to find out why this is happening.

**Reply by UnAfraid** [reply]
Any chance you installed amd64 version of go instead of arm64 by mistake?

**Reply by geertjanvdk** [reply]
Euh.. no :)

**Reply by edosssa** [reply]
+1
